docs/advanced/pooled-instances.rst

```diff
@@ -0,0 +1,241 @@
+        public WorkOperation(ICustomConnection customConnection)
+        {
+            // ...
+        }
+    }
+
+When the current :ref:`Lifetime Scope <lifetime-instance-scope-per-lifetime-scope>` ends, the retrieved instance is returned to the pool.
+
+Resetting Pooled Instances Between Resolves
+-------------------------------------------
+
+With pooled components, there is often a need to do some work to reset the object when it is
+retrieved from or returned to the pool.
+
+Autofac allows a component to be aware of when it is retrieved from or returned to the pool, by
+implementing the ``IPooledComponent`` interface:
+
+.. sourcecode:: csharp
+
+    public class PoolAwareComponent : IPooledComponent
+    {
+        public void OnGetFromPool(IComponentContext context, IEnumerable<Parameter> parameters)
+        {
+            // Called when the component is retrieved from the pool during a resolve operation,
+            // including the first time it is used.
+        }
+
+        public void OnReturnToPool()
+        {
+            // Called when the component is about to be returned to the pool.
+        }
+    }
+
+The ``OnGetFromPool`` method is passed the temporary ``IComponentContext`` of the current resolve
+operation, plus any parameters passed to the resolve.
+
+.. warning:: 
+
+    Any services resolved from the provided ``IComponentContext`` are taken from the **current scope** 
+    accessing the pooled component. This means that any instances you resolve from that ``IComponentContext`` 
+    should be discarded in ``OnReturnToPool`` to prevent memory leaks.
+
+If you cannot modify the component you are pooling, but need to have custom behaviour similar to this,
+you can implement a custom :ref:`Pool Policy <pooled-instances-policies>`.
+
+Pool Capacity
+-------------
+
+Each pooled registration has the notion of a pool capacity.  This capacity defaults to ``Environment.ProcessorCount * 2``,
+but can easily be customised using overloads of the extension methods:
+
+.. sourcecode:: csharp
+
+    // Set a capacity of 100
+    builder.RegisterType<MyCustomConnection>()
+            .As<ICustomConnection>()
+            .PooledInstancePerLifetimeScope(100);
+
+It's important to understand that the capacity of a pool does not place a limit on the number of instances it allocates/activates,
+or can be in use at any one time; instead it limits how many instances are **retained** by the pool.
+
+In practical terms, this means that if your pool capacity is 100, and you currently have 100 instances in use, then
+resolving another instance will just activate a brand new instance of the component, rather than blocking/failing. 
+
+However, if you have 101 instances of the component in use, the next instance that is returned to the pool will be discarded,
+rather than retained. 
+
+In this situation, the ``OnReturnToPool`` method on ``IPooledComponent`` would still be called, but the instance will then immediately be thrown away.
```

Comment:
```suggestion
However, if you have 101 instances of the component in use, the next instance that is returned to the pool will be discarded
rather than retained. In this situation, the ``OnReturnToPool`` method on ``IPooledComponent`` would still be called, but the instance will then immediately be thrown away.
```